### PR TITLE
feat(install): one-line curl|sh installer (installs + wires in one shot)

### PR DESCRIPTION
**Pull request by colinsurprenant**
## What

A `curl | sh` one-line installer, so a discoverer goes from the landing page to **installed and wired** in a single command:

```bash
curl -fsSL https://raw.githubusercontent.com/colinsurprenant/director/main/install.sh | sh
```

`install.sh` detects the platform, resolves the latest release, downloads and sha256-verifies the tarball, stages the binary into `~/.local/bin` and swaps it in with a validated atomic rename, smoke-tests it, then runs `director install` to wire the agent. Options: `--codex` / `--both` / `--no-wire`, `--version`, `--dir` (space or `=` form). Windows users are routed to WSL.

Also switches the **README `## Install`** and the **landing-page CTA** off `go install` to the one-liner, and adds a **shellcheck + `dash -n` CI gate** for the script.

## Why

This closes the "did step 1, forgot step 2" onboarding gap: the previous front door was `go install …@latest` (Go 1.25 toolchain floor) followed by a *separate* `director install` that was easy to forget, with a silent-PATH failure mode if the binary wasn't reachable. The one-liner needs no Go toolchain and leaves the agent wired in one shot. Part of the Phase-0 "seal the funnel" work before any diffusion push (plan: `~/.claude/plans/i-want-to-discuss-synthetic-lightning.md`).

## Review

Ran **ce:code-reviewer** (verdict: approve — "genuinely well-engineered and security-conscious; no Critical or High") and a **codex adversarial pass**. Findings reconciled and addressed:

- Attributable API-failure diagnostics (a piped `fetch | sed` masked network/rate-limit errors; now a guarded fetch-into-var).
- `HOME`-less environments: the `~/.local/bin` default is applied after arg parsing, so `--dir` / `DIRECTOR_INSTALL_DIR` rescue a stripped env instead of aborting under `set -u`.
- Validate-before-replace: stage into the destination dir, smoke-test, then atomic same-fs rename, so an interrupted/bad download can't clobber a working `director`.
- Directory-target guard; extract only the `director` member; `head -1` → `sed 1q`; `${PATH:-}` guard; `--opt=value` form; trailing-slash `--dir`.
- Honest copy: README no longer claims it "puts it on your PATH" (it installs to `~/.local/bin` and warns if that's off PATH); the site caption makes Claude-Code-the-default explicit.

**Intentionally out of scope** (both reviewers agreed acceptable): the checksum verifies download *integrity*, not supply-chain provenance (asset + `checksums.txt` share the GitHub-release trust root the script already relies on via TLS) — a signed-release upgrade (minisign/cosign) is a separate follow-up; full tarball path-traversal hardening is over-engineering for a checksum-verified first-party artifact.

## Testing

Verified end-to-end against the live `v1.6.1` release:
- download + checksum-verify + extract + smoke-test, and a fully-sandboxed install **+ wire** writing the four `_managedBy:"director"` hooks into a temp `settings.json`;
- edge cases: `env -u HOME … --help`, `env -u HOME … --dir`, `--version=`/`--dir=` forms, directory-target guard, bad-option exit code;
- `sh -n`, `dash -n`, and `shellcheck -s sh install.sh` all clean.

No Go code changed, so the §13 gate is unaffected (CI still runs it). The new `installer` CI job uses no `github.event.*` input — no workflow-injection surface.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+		sed -n 's/.*"tag_name"[[:space:]]*:[[:space:]]*"\([^"]*\)".*/\1/p' | sed 1q)"
+	[ -n "$VERSION" ] || die "could not parse a release tag from the GitHub API — set DIRECTOR_VERSION"
+fi
+
+asset="director_${VERSION}_${os}_${arch}.tar.gz"
+base="https://github.com/$REPO/releases/download/$VERSION"
+
+# --- download + verify + extract ----------------------------------------
+tmp="$(mktemp -d)"
+staged="" # same-dir staging copy of the binary; removed if we die mid-install
```
[the remark this answers is not in the document]
Comment (colinsurprenant) [reply install.sh:161]: Verified: bare `mktemp -d` works on current macOS (15.7.5, confirmed live) and on all supported platforms — GNU, BSD/macOS, and busybox — so the erroring behavior doesn't reproduce here. Switched to an explicit `${TMPDIR:-/tmp}/director-install.XXXXXX` template anyway: it's the universally-portable form at zero cost, which is the right posture for a front-door installer running on unknown machines.

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+# The ~/.local/bin default is applied AFTER arg parsing so that --dir /
+# DIRECTOR_INSTALL_DIR work even when HOME is unset (stripped envs, cron, sudo).
+INSTALL_DIR="${DIRECTOR_INSTALL_DIR:-}"
+
+# --- output --------------------------------------------------------------
+# Progress goes to stderr; the wired `director install` writes its own
+# confirmations to stdout.
+info() { printf '%s\n' "$*" >&2; }
+err() { printf 'director install: %s\n' "$*" >&2; }
+die() {
```
[the remark this answers is not in the document]
Comment (colinsurprenant) [reply install.sh:46]: Fixed: installer-script errors now use a `director installer:` prefix, distinct from the `director install` subcommand's own output — so the wire-failure path (where both appear) no longer conflates the two.
